pal/wasm/wasm-minigame.ts

```diff
@@ -44,6 +44,53 @@ export function fetchBuffer (binaryUrl: string): Promise<ArrayBuffer> {
     });
 }
 
+function loadSubpackage (name: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+        if (minigame.loadSubpackage) {
+            minigame.loadSubpackage({
+                name,
+                success () {
+                    resolve();
+                },
+                fail (err) {
+                    reject(err);
+                },
+            });
+        } else {
+            reject(new Error(`Subpackage is not supported on this platform`));
+        }
+    });
+}
+
+let isWasmModuleReady = false;
```

Comment:
Why need this variable? Just to cache result if `ensureWasmModuleReady `  is invoked more than one time?

Reply:
on minigame platforms, we only need to load subpackage once
some platforms would cache the load result,
 some won't ,on these platforms they would execute the entry script twice , which doesn't meet our expectation

Reply:
I optmized this with a promise to load module
https://github.com/cocos/cocos-engine/pull/15631/commits/5487b1e886397566cc8bea54b3667a40d5adcec7